src/catalog/src/durable.rs

```diff
@@ -311,6 +314,100 @@ pub trait DurableCatalogState: ReadOnlyDurableCatalogState {
         let id = id.into_element();
         Ok(ReplicaId::User(id))
     }
+
+    /// Migrates storage stash data from `other` to `self`.
+    async fn migrate_storage_state_from(
+        &mut self,
+        other: &mut Box<dyn DurableCatalogState>,
+    ) -> Result<(), CatalogError> {
+        use std::collections::BTreeMap;
+
+        use mz_proto::RustType;
+        use mz_storage_client::controller::StorageTxn;
+
+        let tombstoned = other.get_tombstone().await?;
+        if tombstoned == Some(true) {
+            return Ok(());
+        }
```

Comment:
I don't think you can re-use this existing tombstone, it's specifically used for the stash catalog to persist catalog migration. You'll have to create a new tombstone-like value.